### PR TITLE
ReturnBookUI

### PR DESCRIPTION
**Pull request by elliotmitchum**
Refactors `ReturnBookUI` class.

Criteria:

- Evaluate against the class diagram
- Evaluate against the code guidelines
- Project builds and runs

### CODE REVIEW

**Comment by elliotmitchum** [comment]
@haynescrossley This is ready for review.

**Comment by amrmosit** [comment]
@haynescrossley  PR42 is ready for review. 
Code is still operational

File: src/library/returnBook/ReturnBookUI.java [hunk 1]
```diff
@@ -1,101 +1,100 @@
 package library.returnBook;
+
 import java.util.Scanner;
 
 
 public class ReturnBookUI {
```
Comment (haynescrossley) [inline src/library/returnBook/ReturnBookUI.java]: #PR42 - D1: Please rename class to ReturnItemUI as per class diagram.

File: src/library/returnBook/ReturnBookUI.java [hunk 2]
```diff
@@ -1,101 +1,100 @@
...
 
-    private enum uI_sTaTe { INITIALISED, READY, INSPECTING, COMPLETED };
+    private enum UiState {INITIALISED, READY, INSPECTING, COMPLETED};
```
Comment (haynescrossley) [inline src/library/returnBook/ReturnBookUI.java]: #PR42 - D2: Please rename/refactor variable to ReturnItemUIState as per class diagram

File: src/library/returnBook/ReturnBookUI.java [hunk 3]
```diff
@@ -1,101 +1,100 @@
...
+
+    private ReturnItemControl control;
+
+    private Scanner input;
```
Comment (haynescrossley) [inline src/library/returnBook/ReturnBookUI.java]: #PR42 - D3: Please rename/refactor Scanner variable to 'scanner' as per class diagram

File: src/library/returnBook/ReturnBookUI.java [hunk 4]
```diff
@@ -1,101 +1,100 @@
...
 
-    private ReturnItemControl CoNtRoL;
-    private Scanner iNpUt;
-    private uI_sTaTe StATe;
+    private UiState state;
```
Comment (haynescrossley) [inline src/library/returnBook/ReturnBookUI.java]: #PR42 - D4: Please rename/refactor ReturnItemUIState variable 'state' to 'uiState' as per class diagram

File: src/library/returnBook/ReturnBookUI.java [hunk 5]
```diff
@@ -1,101 +1,100 @@
...
 
-    
-    public ReturnBookUI(ReturnItemControl cOnTrOL) {
-        this.CoNtRoL = cOnTrOL;
-        iNpUt = new Scanner(System.in);
-        StATe = uI_sTaTe.INITIALISED;
-        cOnTrOL.setUI(this);
+
+    public ReturnBookUI(ReturnItemControl control) {
```
Comment (haynescrossley) [inline src/library/returnBook/ReturnBookUI.java]: #PR - D5: Please rename constructor to ReturnItemUI (probably will be fixed when refactoring class/file name)

File: src/library/returnBook/ReturnBookUI.java [hunk 6]
```diff
@@ -1,101 +1,100 @@
...
-            default:
-                DiSpLaYoUtPuT("Unhandled state");
-                throw new RuntimeException("ReturnBookUI : unhandled state :" + StATe);            
+                    break;
+
+                case INSPECTING:
+                    String answer = getInput("Is book damaged? (Y/N): ");
+                    boolean isDamaged = false;
+                    if (answer.toUpperCase().equals("Y"))
+                        isDamaged = true;
```
Comment (haynescrossley) [inline src/library/returnBook/ReturnBookUI.java]: #PR42 - D6 : Please add brackets to if statement as per CSG (anywhere curly brackets can be used they MUST be used)

**Review by haynescrossley — CHANGES_REQUESTED** [review]
@elliotmitchum Please resolve changes as per my comments. Let me know if there's any issues.

File: src/library/returnBook/ReturnBookUI.java (hunk 1 above)
Comment (elliotmitchum) [reply src/library/returnBook/ReturnBookUI.java]: https://github.com/elliotmitchum/ITC205/pull/42/commits/8b325666fce4f04d4977b7d4dd50069c5cf81a7c

File: src/library/returnBook/ReturnBookUI.java (hunk 2 above)
Comment (elliotmitchum) [reply src/library/returnBook/ReturnBookUI.java]: https://github.com/elliotmitchum/ITC205/pull/42/commits/c565c7508e76f540caef5711aa968f5a9487ae2f

File: src/library/returnBook/ReturnBookUI.java (hunk 3 above)
Comment (elliotmitchum) [reply src/library/returnBook/ReturnBookUI.java]: https://github.com/elliotmitchum/ITC205/pull/42/commits/6e206a82e1b6db14ba97f81a1c51dd5b4773c823

File: src/library/returnBook/ReturnBookUI.java (hunk 6 above)
Comment (elliotmitchum) [reply src/library/returnBook/ReturnBookUI.java]: https://github.com/elliotmitchum/ITC205/pull/42/commits/20d8573a2348360d7921fdb921d86190bc79da3f

File: src/library/returnBook/ReturnBookUI.java (hunk 5 above)
Comment (elliotmitchum) [reply src/library/returnBook/ReturnBookUI.java]: https://github.com/elliotmitchum/ITC205/pull/42/commits/8b325666fce4f04d4977b7d4dd50069c5cf81a7c

**Comment by elliotmitchum** [comment]
@haynescrossley I have updated with your feedback and is ready for re-review

File: src/library/returnBook/ReturnBookUI.java (hunk 4 above)
Comment (haynescrossley) [reply src/library/returnBook/ReturnBookUI.java]: [110d34d](https://github.com/elliotmitchum/ITC205/pull/42/commits/110d34d17174af49be196ef13a74b9516fef2a17)

**Review by haynescrossley — APPROVED** [review]
@amrmosit All 'defects' resolved. I'm approving this PR. Ready for you to check execution and merge.

**Comment by amrmosit** [comment]
Merged successfully

**Pull request merged**